code/forms/JsonField.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/**
+ * Class JsonField combines form inputs into a key-value pair
+ */
+class JsonField extends CompositeField
+{
+    public function __construct($name, $children = null)
+    {
+        $this->setName($name);
+
+        if ($children instanceof FieldList) {
+            foreach ($children as $child) {
+                $child->setName($this->getName() . "[" . $child->getName() . "]");
+            }
+        } elseif (is_array($children)) {
+            foreach ($children as $child) {
+                $child->setName($this->getName() . "[" . $child->getName() . "]");
+            }
+        } else {
+            $children = is_array(func_get_args()) ? func_get_args() : array();
+            if (!empty($children)) {
+                array_shift($children);
+            }
+            foreach ($children as $child) {
+                $child->setName($this->getName() . "[" . $child->getName() . "]");
```

Comment:
We've got this formatting line (adding square brackets) three times - maybe it should be in a method